apps/src/lib/client/tm_jsonrpc_client.rs

```diff
@@ -0,0 +1,256 @@
+        }
+    }
+
+    /// The body of a json rpc request
+    #[derive(Serialize)]
+    pub struct Request {
+        /// Method name
+        pub method: String,
+        /// parameters to give the method
+        params: EventParams,
+        /// ID of the request
+        id: u8,
+    }
+
+    impl From<EventParams> for Request {
+        fn from(params: EventParams) -> Self {
+            Request {
+                method: "events".into(),
+                params,
+                id: 1,
+            }
+        }
+    }
+
+    /// The response we get back from Tendermint
+    #[derive(Serialize, Deserialize)]
+    pub struct Response {
+        /// JSON-RPC version
+        jsonrpc: String,
+        /// Identifier included in request
+        id: u8,
+        /// Results of request (if successful)
+        result: Option<EventReply>,
+        /// Error message if unsuccessful
+        error: Option<tendermint_rpc::response_error::ResponseError>,
+    }
+
+    impl Response {
+        /// Convert the response into a result type
+        pub fn into_result(self) -> Result<EventReply, Error> {
+            if let Some(e) = self.error {
+                Err(Error::Rpc(e))
+            } else if let Some(result) = self.result {
+                Ok(result)
+            } else {
+                Err(Error::MalformedJson)
+            }
+        }
+    }
+
+    /// Holds bytes returned in response to curl request
+    #[derive(Default)]
+    pub struct Collector(Vec<u8>);
+
+    impl Handler for Collector {
+        fn write(&mut self, data: &[u8]) -> Result<usize, WriteError> {
+            self.0.extend_from_slice(data);
+            Ok(data.len())
+        }
+    }
+
+    /// The RPC client
+    pub struct Client<'a> {
+        /// The actual curl client
+        inner: Easy2<Collector>,
+        /// Url to send requests to
+        url: &'a str,
+        /// The request body
+        request: Request,
+        /// The hash of the tx whose corresponding event is being searched for.
+        hash: &'a str,
+    }
+
+    impl<'a> Deref for Client<'a> {
+        type Target = Easy2<Collector>;
+
+        fn deref(&self) -> &Self::Target {
+            &self.inner
+        }
+    }
+
+    impl<'a> DerefMut for Client<'a> {
+        fn deref_mut(&mut self) -> &mut Self::Target {
+            &mut self.inner
+        }
+    }
+
+    impl<'a> Client<'a> {
+        /// Create a new client
+        pub fn new(url: &'a str, request: Request, hash: &'a str) -> Self {
+            let mut client = Self {
+                inner: Easy2::new(Collector::default()),
+                url,
+                request,
+                hash,
+            };
+            client.reset();
+            client
+        }
+
+        /// Send a request to Tendermint
+        ///
+        /// Takes the 10 newest block header events and searches for
+        /// the relevant event among them.
+        pub fn send(&mut self) -> Result<TxResponse, Error> {
```

Comment:
There are more advanced features of the event log that involves paging back through chunks of logs. I originally was doing something similar. I decided to leave this in place in case we use these features again in the future.